app/core/src/test/java/com/fsck/k9/helper/ListUnsubscribeHelperTest.kt

```diff
@@ -0,0 +1,89 @@
+package com.fsck.k9.helper
+
+import com.fsck.k9.K9RobolectricTest
+import com.fsck.k9.mail.internet.MimeMessage
+import org.junit.Assert
+import org.junit.Test
+
+class ListUnsubscribeHelperTest : K9RobolectricTest() {
+    @Test
+    fun getListUnsubscribeUrl_shouldAcceptMailto() {
+        val message = buildMimeMessageWithListUnsubscribeValue(
+            "<mailto:unsubscribe@example.com>"
+        )
+        val result = ListUnsubscribeHelper.getPreferredListUnsubscribeUri(message)
+        Assert.assertTrue(result is MailtoUnsubscribeUri)
+        Assert.assertEquals("mailto:unsubscribe@example.com", result!!.uri.toString())
```

Comment:
In new code we're using [Truth](https://truth.dev/) for assertions. Combined with kotlin.test's `assertNotNull` we end up with:

```kotlin
assertNotNull(result) { unsubscribeUri ->
    assertThat(unsubscribeUri).isInstanceOf(MailtoUnsubscribeUri::class.java)
    assertThat(unsubscribeUri.uri.toString()).isEqualTo("mailto:unsubscribe@example.com")
}
```

But because we're using data classes we could also do this:

```kotlin
assertThat(result).isEqualTo(MailtoUnsubscribeUri("mailto:unsubscribe@example.com".toUri()))
```
```

Reply:
I went for the first example, because I think the second example doesn't actually test if the correct UnsubscribeUri type is returned, because both versions have toUri().

Reply:
They are tests. If you're not sure they're testing the right thing, try to make them fail.

Reply:
Ah, yeah, that's my bad. I read your example as:

```
assertThat(result).isEqualTo(MailtoUnsubscribeUri("mailto:unsubscribe@example.com").toUri())
```

instead of

```
assertThat(result).isEqualTo(MailtoUnsubscribeUri("mailto:unsubscribe@example.com".toUri()))
```

Long day at work today :)